From 179bc0d4af1e58a90f28cdec58a71c44fcb7f18f Mon Sep 17 00:00:00 2001
From: Moritz Fischer <moritz.fischer@ettus.com>
Date: Fri, 27 Apr 2018 15:32:51 -0700
Subject: [PATCH] ni: zynq: Mender.io support for NI Ettus Research E31x series
 SDRs

This involves adding bootcount in environment, as well as modifying
the default bootconfiguration to run ${mender_setup} first.
---
 configs/ni_e31x_sg1_defconfig |  1 +
 configs/ni_e31x_sg3_defconfig |  1 +
 include/configs/ni_e31x.h     | 11 +++++------
 3 files changed, 7 insertions(+), 6 deletions(-)

diff --git a/configs/ni_e31x_sg1_defconfig b/configs/ni_e31x_sg1_defconfig
index 66a391b703..712d58cef0 100644
--- a/configs/ni_e31x_sg1_defconfig
+++ b/configs/ni_e31x_sg1_defconfig
@@ -11,6 +11,7 @@ CONFIG_DEFAULT_DEVICE_TREE="zynq-ni-e31x-sg1"
 CONFIG_DEBUG_UART=y
 CONFIG_FIT=y
 CONFIG_FIT_SIGNATURE=y
+CONFIG_SYS_EXTRA_OPTIONS="BOOTCOUNT_LIMIT,BOOTCOUNT_ENV"
 CONFIG_BOOTDELAY=0
 # CONFIG_DISPLAY_CPUINFO is not set
 CONFIG_SPL=y
diff --git a/configs/ni_e31x_sg3_defconfig b/configs/ni_e31x_sg3_defconfig
index 9a3c25846a..7a1e12a1be 100644
--- a/configs/ni_e31x_sg3_defconfig
+++ b/configs/ni_e31x_sg3_defconfig
@@ -11,6 +11,7 @@ CONFIG_DEFAULT_DEVICE_TREE="zynq-ni-e31x-sg3"
 CONFIG_DEBUG_UART=y
 CONFIG_FIT=y
 CONFIG_FIT_SIGNATURE=y
+CONFIG_SYS_EXTRA_OPTIONS="BOOTCOUNT_LIMIT,BOOTCOUNT_ENV"
 CONFIG_BOOTDELAY=0
 # CONFIG_DISPLAY_CPUINFO is not set
 CONFIG_SPL=y
diff --git a/include/configs/ni_e31x.h b/include/configs/ni_e31x.h
index 8c60447097..4fad5dd3ed 100644
--- a/include/configs/ni_e31x.h
+++ b/include/configs/ni_e31x.h
@@ -10,11 +10,9 @@
 #ifndef __CONFIG_NI_E31X_H
 #define __CONFIG_NI_E31X_H

+#define CONFIG_ENV_IS_NOWHERE
 #include <configs/zynq-common.h>
-
-#define CONFIG_SYS_MMC_ENV_DEV 0
-#define CONFIG_ENV_SECT_SIZE           CONFIG_ENV_SIZE
-#define CONFIG_ENV_OFFSET              0xE0000
+#undef CONFIG_ENV_IS_NOWHERE

 #undef CONFIG_EXTRA_ENV_SETTINGS
 #define CONFIG_EXTRA_ENV_SETTINGS	\
@@ -31,8 +29,9 @@
 		"echo Loading FIT to RAM via DHCP/TFTP && " \
 		"dhcp ${load_addr} ${tftproot}/${fit_image} && "\
 		"bootm ${load_addr}#conf@zynq-ni-${mboard}.dtb\0" \
-	"sdboot=echo Copying FIT from SD to RAM... && " \
-		"ext4load mmc 0:2 ${load_addr} ${fit_image} && " \
+	"sdboot=run mender_setup; " \
+		"echo Copying FIT from SD to RAM... && " \
+		"ext4load ${mender_uboot_root} ${load_addr} ${fit_image} && " \
 		"bootm ${load_addr}#conf@zynq-ni-${mboard}.dtb\0" \
 	"jtagboot=echo TFTPing FIT to RAM... && " \
 		"tftpboot ${load_addr} ${fit_image} && " \
--
2.17.1
